tests/e2e/gc_file_share_negative.go

```diff
@@ -0,0 +1,147 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	fnodes "k8s.io/kubernetes/test/e2e/framework/node"
+	fpv "k8s.io/kubernetes/test/e2e/framework/pv"
+)
+
+var _ = ginkgo.Describe("[csi-guest] File Share on Non File Service enabled setups", func() {
+	f := framework.NewDefaultFramework("e2e-file-volumes")
+
+	var (
+		client            clientset.Interface
+		namespace         string
+		scParameters      map[string]string
+		storagePolicyName string
+	)
+
+	ginkgo.BeforeEach(func() {
+		client = f.ClientSet
+		namespace = getNamespaceToRunTests(f)
+		nodeList, err := fnodes.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+		bootstrap()
+		scParameters = make(map[string]string)
+		storagePolicyName = GetAndExpectStringEnvVar(envStoragePolicyNameForSharedDatastores)
+		svcClient, svNamespace := getSvcClientAndNamespace()
+		setResourceQuota(svcClient, svNamespace, rqLimit)
+	})
+
+	ginkgo.AfterEach(func() {
+		svcClient, svNamespace := getSvcClientAndNamespace()
+		setResourceQuota(svcClient, svNamespace, defaultrqLimit)
+	})
+
+	/*
+		Create and check if File share creation fails on non file service enabled setup
+		1. Create a SC
+		2. Create a PVC with "ReadWriteMany" access mode using the SC from above in GC
+		3. Expect the PVC creation fails
+		4. Verify the error message returned on PVC failure is correct
+		5. Delete PVC in GC
+	*/
+
+	ginkgo.It("Verify File share creation fails on non-file service enabled setup - ReadWriteMany", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		var storageclasspvc *storagev1.StorageClass
+		var pvclaim *v1.PersistentVolumeClaim
+
+		var err error
+		defaultDatastore = getDefaultDatastore(ctx)
+
+		ginkgo.By("Creating Storage Class and PVC")
+		scParameters[svStorageClassName] = storagePolicyName
+
+		ginkgo.By("Creating a Storage class and PVC")
+		storageclasspvc, pvclaim, err = createPVCAndStorageClass(client,
+			namespace, nil, scParameters, diskSize, nil, "", false, v1.ReadWriteMany)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			err = client.StorageV1().StorageClasses().Delete(ctx, storageclasspvc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Expect claim to fail")
+		err = fpv.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, client,
```

Comment:
instead of waiting after timeout, poll for the failure from the events
this will be faster as well

Reply:
Events are dynamic - using an increased bound timeout instead